assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -155,14 +172,95 @@ export const reducer = createReducer( ( state, { type, payload } ) => {
 			break;
 		}
 
+		case RECEIVE_QUEUED_NOTIFICATIONS: {
+			state.queuedNotifications = payload.queuedNotifications;
+			break;
+		}
+
 		default:
 			break;
 	}
 } );
 
-export const resolvers = {};
+export const resolvers = {
+	*getQueuedNotifications( viewContext ) {
+		const registry = yield Data.commonActions.getRegistry();
+
+		const notifications = registry
+			.select( CORE_NOTIFICATIONS )
+			.getNotifications();
+
+		const filteredNotifications = Object.values( notifications ).filter(
+			( notification ) => {
+				if ( ! notification.viewContexts.includes( viewContext ) ) {
+					return false;
+				}
+
+				if (
+					!! notification.isDismissible &&
+					registry
+						.select( CORE_NOTIFICATIONS )
+						.isNotificationDismissed( notification.id )
```

Comment:
Will this work if dismissed items are not resolved yet? This call will return `undefined`, right? We probably need to call `resolveSelect` to get dismissed items before we can filter notifications.